python/gink/impl/key_set.py

```diff
@@ -0,0 +1,222 @@
+""" Contains the key set class definition """
+
+from typing import Optional, Iterable
+
+from .database import Database
+from .muid import Muid
+
+from .muid import Muid
+from .database import Database
+from .container import Container
+from .coding import KEY_SET, deletion, decode_key, inclusion
+from .bundler import Bundler
+from .typedefs import UserKey, GenericTimestamp
+from .builders import Behavior
+
+class KeySet(Container):
+    _missing = object()
+    BEHAVIOR = KEY_SET
+
+    def __init__(self, root: Optional[bool] = None, bundler: Optional[Bundler] = None, contents = None, 
+                 muid: Optional[Muid] = None, database = None, comment: Optional[str] = None):
+        """
+        Constructor for a set proxy.
+        
+        muid: the global id of this set, created on the fly if None
+        db: database to send commits through, or last db instance created if None
+        """
+        if root:
+            muid = Muid(-1, -1, KEY_SET)
+        database = database or Database.get_last()
+        immediate = False
+        if bundler is None:
+            immediate = True
+            bundler = Bundler(comment)
+        if muid is None:
+            muid = Container._create(KEY_SET, database=database, bundler=bundler)
+        elif muid.timestamp > 0 and contents:
+            # TODO [P3] check the store to make sure that the container is defined and compatible (possibly for set as well?)
+            pass
+        Container.__init__(self, muid=muid, database=database)
+        if contents:
+            self.clear(bundler=bundler)
+            # self.update(contents, bundler=bundler)
+        if immediate and len(bundler):
+            self._database.commit(bundler)
+
+    def add(self, key: UserKey, *, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Adds a specified value to the key set """
+        return self._add_entry(key=key, value=inclusion, bundler=bundler, comment=comment)
+    
+    def update(self, keys: Iterable[UserKey], bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Adds multiple specified values to the key set """
+        # Probably best to initialize a bundler here
```

Comment:
A single call to update should make all of it's changes in one transaction (bundle).